src/NinjaTools.FluentMockServer.API/Downstream/RequestMapper.cs

```diff
@@ -0,0 +1,167 @@
+    public interface IMapper<in TFrom, TTo>
+    {
+        /// <summary>
+        /// TODO: Not store in IMapper -> contradicts  SRP
+        /// </summary>
+        Func<TTo, TTo, bool> EqualityComparerDelegate { get; }
+
+        /// <summary>
+        /// Maps the <paramref name="from"/> to a <typeparam name="TTo" />.
+        /// </summary>
+        /// <param name="from">The <see cref="TFrom"/>.</param>
+        /// <returns>
+        /// Returns a <see cref="Response{TTo}"/> containing the mapping result or any <see cref="Error"/> that occured.
+        /// </returns>
+        Task<Response<TTo>> MapAsync(TFrom from);
+    }
+
+    /// <inheritdoc />
+    public interface IRequestMapper : IMapper<HttpRequest, HttpRequestMessage>
+    {
+    }
+
+    /// <inheritdoc />
+    public class RequestMapper : IRequestMapper
+    {
+        /// <summary>
+        /// The default instance of <see cref="RequestMapper"/>.
+        /// </summary>
+        public static readonly RequestMapper Default = new RequestMapper();
+
+        /// <summary>
+        /// Headers that are excluded from the mapping.
+        /// </summary>
+        public readonly string[] UnsupportedHeaders = {"host"};
+
+        /// <inheritdoc />
+        public Func<HttpRequestMessage, HttpRequestMessage, bool> EqualityComparerDelegate { get; } = (a, b) => HttpRequestMessageEqualityComparer.Default.Equals(a, b);
+
+        /// <inheritdoc />
+        public async Task<Response<HttpRequestMessage>> MapAsync(HttpRequest request)
+        {
+            try
+            {
+                var requestMessage = new HttpRequestMessage
+                {
+                    Content = await MapContentAsync(request),
+                    Method = MapMethod(request),
+                    RequestUri = MapUri(request)
+                };
+
+                MapHeaders(request, requestMessage);
+
+                return new OkResponse<HttpRequestMessage>(requestMessage);
+            }
+            catch (Exception ex)
+            {
+                return new ErrorResponse<HttpRequestMessage>(new RequestMappingError(ex));
+            }
+        }
+
+        private async Task<HttpContent> MapContentAsync([NotNull] HttpRequest request)
+        {
+            if (request.Body == null || (request.Body.CanSeek && request.Body.Length <= 0))
+            {
+                return null;
+            }
+
+            if (request.Body != null)
+            {
+                request.EnableBuffering();
+            }
+   
+            
+            // Never change this to StreamContent again, I forgot it doesnt work in #464.
+            var content = new ByteArrayContent(await ToByteArray(request.Body));
+            content.Headers.TryAddWithoutValidation("Content-Type", new[] {request.ContentType});
+
+            // if (!string.IsNullOrEmpty(request.ContentType))
+            //     content.Headers.TryAddWithoutValidation("Content-Type", new[] { request.ContentType });
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/app/git_36/NinjaTools.FluentMockServer/pullRequest?prid=4722826)